src/lib/connectors/shared/remote-debugging-connector.ts

```diff
@@ -618,61 +670,81 @@ export class Connector implements IConnector {
         }
     }
 
-    /** Initiates all the process */
-    private onLoadEventFired(callback: Function): Function {
-        return () => {
-            const { DOM } = this._client;
-            const event: IEvent = { resource: this._finalHref };
+    /** Stops processing any network requests and returns once all the
+     * pending requests have been processed or `maxLoadWaitTime`,
+     * whichever happens first. */
+    // TODO: find a cleaner way of doing this
+    private waitForNetwork: Function = promisify((callback) => {
```

Comment:
IMO, it is more clear if you return a `new Promise` instead of promisify a function